src/WorldMap.cpp

```diff
@@ -40,3 +235,200 @@ void WorldMap::clearMap()
 {
 	m_tileLayers.clear();
 }
+
+void WorldMap::addTileSheet(const std::string& tileSheetPath, const TileSheetDetails & tileSheetDetails)
+{
+	assert(!hasTileLayer(tileSheetDetails.m_name));
+	m_tileSheets.emplace(tileSheetDetails.m_name, TileSheet(tileSheetDetails, tileSheetPath));
+}
+
+void WorldMap::parseTileMap(const TiXmlElement & root, const LevelDetails & levelDetails)
+{
+	const int tileLayerCount = getNumberOfTileLayers(root);
+	const TiXmlElement* tileLayerNode = findNode(root, "layer"); //Get the first layer node
+	assert(tileLayerNode);
+	for (int i = 1; i <= tileLayerCount; ++i)
+	{
+		//If node in question is a layer 
+		if (tileLayerNode->Value() == std::string("layer") &&
+			tileLayerNode->Attribute("name") == std::string("Tile Layer " + std::to_string(i)))
+		{
+			std::string tileLayerName = tileLayerNode->Attribute("name");
+			//assert(tileLayerNode->Attribute("tilesheetname"));
+			std::string tileSheetName = "tilesheet_complete";
+
+			if (tileLayerNode->FirstChildElement()->Value() == std::string("data") ||
+				tileLayerNode->FirstChildElement()->NextSiblingElement() != 0 && tileLayerNode->FirstChildElement()->NextSiblingElement()->Value() == std::string("data"))
```

Comment:
Seems fishy that we only pull data from either the first or second child here.  Should be a loop.